### DISCUSSION TITLE
Where to find the cell area for the LAM outputs

### DISCUSSION THREAD

**Discussion by XiaSun-Atmos** [ufs-community/ufs-srweather-app#899] [Q&A] [answered]
Hi,
I would like to ask if it's possible to extrat the `cell areas` for SRW output which used lambert conformal projection.
I see there is `area` variable in `grid_spec.nc` file, but due to the conversion of ESG grid to wirte-component grids, the dimensions in phyf*.nc is generally smaller than dimensions in grid_spec.nc. I cannot match the lat and lons in grid_spec.nc to match those in phyf*.nc, thus the area variable.
My question is how to find the cell areas corresponding to grids in phyf*.nc? I am using those cell areas to calculate area-average weights to do further interpolation.
Thanks,

**Comment by gsketefian** [comment]
Hi @XiaSun-Atmos, you're right, the write-component generation code in the weather model does not include the areas of the write-component grid cells in the output files (`dynf###.nc` and `phyf###.nc`).  This is probably something others have done before and a formula for it can be derived from the mapping equations, but I personally don't know it.  I will tag @DusanJovic-NOAA since I think he's the author of the write-component code in the WM.  @DusanJovic-NOAA, would it be possible to add the cell areas in either `dynf###.nc` or `phyf###.nc`?  Also, do you know the formula to use for calculating them (so @XiaSun-Atmos can calculate them if it will take a long time to put into the WM)?  Thanks.

**Reply by DusanJovic-NOAA** [reply]
@gsketefian I do not know the formula for the grid cell area.

**Reply by XiaSun-Atmos** [reply]
Thanks @gsketefian and @DusanJovic-NOAA .
Does that mean it is not possible to have the cell areas for UFS LAM outputs with Lambert conformal projection at this time?
Thanks,

**Comment by gsketefian** [comment, marked as answer]
@XiaSun-Atmos I thought about this a bit more yesterday, and I think it's not too hard to calculate the cell areas.  I think all you have to do is take the lat and lon coordinates of the grid (which are available in say `dynf000.nc`) and, for each cell, calculate a dlat and dlon.  Then use the usual formulas to get the physical area:
```
  ds_lon (distance in easting direction) = rad_earth*cos(lat)*dlon
  ds_lat (distance in northing direction) = rad_earth*dlat
  dA = ds_lon*ds_lat
```
You'll need the cell face center coordinates to calculate dlon and dlat for each cell, i.e.
```
dlon(i,j) = lon(i+1/2,j) - lon(i-1/2,j)
dlat(i,j) = lat(i,j+1/2) - lat(i,j-1/2)
```
where lon(i+1/2,j) is the longitude of the center of the east face of cell (i,j), and lat(i,j+1/2) is the latitude of the center of the north face of cell (i,j).  In `dynf000.nc`, I believe both the lat-lon coordinates (`lon` and `lat`) and the lambert coordinates (`grid_xt` and `grid_yt`) are for the cell centers, i.e. they are at (i,j).  Depending on how accurately you want to calculate the areas, you can either just average these to get values at cell face centers, or you can first calculate the lambert coordinates of the face centers from the lambert coordinates of the centers (`grid_xt` and `grid_yt`) and then transform them to lat-lon coordinates yourself.  Once you have dlon and dlat for each cell (and you also have lat at the cell center), you can use the formulas above to get the areas dA.

I also recommend plotting the write-component grid in lat-lon space just to make sure things look reasonable.

**Reply by XiaSun-Atmos** [reply]
This is great! Thank you @gsketefian for the detailed information. I will give it a try based on your approach.
